<?xml version="1.0" ?>
<launch>
	<param name="use_sim_time" value="false" />
	
	<!-- arguments -->
	<arg name="gui" default="false" />

	<!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
	<include file="$(find hdt_rlb_moveit)/launch/planning_context.launch">
		<arg name="load_robot_description" value="true"/>
	</include>
	
	<!-- load urdf, start state publishers -->
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
		<param name="use_gui" value="$(arg gui)" />
		<rosparam param="/source_list">[/hdt_adroit_coms/joint_cmd]</rosparam>
	</node>

	<!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
	<include file="$(find hdt_rlb_moveit)/launch/move_group.launch">
		<arg name="allow_trajectory_execution" value="false"/>  
		<arg name="fake_execution" value="false"/>
	</include>

	<!-- launch joystick -->
	<node name="xbox_joystick" pkg="joy" type="joy_node">
		<param name="dev" value="/dev/input/js0" />
		<param name="deadzone" value="0.15" />
	</node>

	<!-- launch hocu -->
	<node name="hocu_driver" pkg="hdt_hocu_driver" type="hocu_driver">
		<param name="port" value="8766" />
	</node>

	<!-- launch coms node -->
	<include file="$(find hdt_adroit_driver)/launch/hdt_adroit_coms.launch">
		<arg name="coms_interface" value="loopback" />
		<arg name="ip_address" value="192.168.1.45" />
		<arg name="port" value="8765" />
		<arg name="impedance_enabled" value="false" />
		<arg name="console_enabled" value="false" />
	</include>
	
	<!-- launch rviz -->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find hdt_rlb_description)/launch/hdt_rlb.rviz"/>

	<!-- launch rlb control node -->
	<node name="rlb_control" pkg="hdt_rlb_control"  type="rlb_control" respawn="false" output="screen">
	</node>
</launch>

